guides/source/active_record_callbacks.md

```diff
@@ -837,58 +837,59 @@ lead to invalid data.
 [`upsert_all`]:
     https://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-upsert_all
 
-Suppressing Callbacks
----------------------
+Suppressing Saving
+------------------
 
-In certain scenarios, you may need to temporarily prevent certain callbacks from
-being executed within your Rails application. This can be useful when you want
-to skip specific actions during certain operations without permanently disabling
-the callbacks.
+In certain scenarios, you may need to temporarily prevent records from being saved within your Rails application.
+This can be useful to skip the callbacks that would run if the record is saved successfully.
 
-Rails provides a mechanism for suppressing callbacks using the
-[`ActiveRecord::Suppressor`
-module](https://api.rubyonrails.org/classes/ActiveRecord/Suppressor.html). By
-using this module, you can wrap a block of code where you want to suppress
-callbacks, ensuring that they are not executed during that specific operation.
+Rails provides a mechanism to prevent saving records using the [`ActiveRecord::Suppressor` module]
+(https://api.rubyonrails.org/classes/ActiveRecord/Suppressor.html).
+By using this module, you can wrap a block of code where you want to avoid saving records and running their save callbacks.
 
-Let's consider a scenario where we have a `User` model with a callback that
-sends a welcome email to new users after they sign up. However, there might be
-cases where we want to create a user without sending the welcome email, such as
-during seeding the database with test data.
+Let's consider a scenario where we have a `User` `has_one` `Notification`.
+Creating a `User` will automatically create a `Notification` record as well.
+Saving a `Notification`, in turn, will trigger a callback that sends an email.
 
 ```ruby
 class User < ApplicationRecord
-  after_create :send_welcome_email
+  has_one :notification
+  after_create :create_notification
```

Comment:
This snippet should include the definition of the callback that creates a notification to make it more clear what's happening.

```suggestion
  after_create :create_welcome_notification
  
  def create_welcome_notification
    notifications.create(event: 'signed_up')
  end
```

Reply:
Just to be sure, we're on the same page:`create_notification` is defined by `has_one :notification`, so there's no missing method definition in my snippet.

I'm not sure if expanding the example makes it clearer.

Reply:
Ah right, #create_notification is generated by the association, I did overlook that. I still think more descriptive names/realistic modeling improve readability.

Reply:
Making the Notification creating more explicit as grncdr suggests makes sense to me.